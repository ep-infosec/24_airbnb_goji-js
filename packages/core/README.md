# Goji Core

![npm (scoped)](https://img.shields.io/npm/v/@goji/core)

For more detailed, see [Goji official docs](https://goji.js.org).
