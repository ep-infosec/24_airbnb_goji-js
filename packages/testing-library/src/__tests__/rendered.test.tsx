import React from 'react';
import { View, Button, Text } from '@goji/core';
import { render, within } from '..';

describe('rendered', () => {
  test('debug', () => {
    const App = () => (
      <View testID="a">
        <Button onTap={() => {}}>hello</Button>
        <Text className="a b c">world</Text>
      </View>
    );

    const wrapper = render(<App />);
    const log = jest.spyOn(console, 'log').mockImplementation(() => {});
    wrapper.debug(wrapper.getByTestId('a'));
    expect(log).toHaveBeenCalledTimes(1);
    const output = log.mock.calls[0][0];
    expect(typeof output).toBe('string');
    expect(output).toMatchSnapshot();
    log.mockRestore();
  });

  test('within', () => {
    const App = () => (
      <>
        <View testID="a">
          <View testID="z" />
        </View>
        <View testID="b" />
      </>
    );

    const wrapper = render(<App />);
    expect(wrapper.getAllByTestId('z')).toHaveLength(1);
    const viewA = wrapper.getByTestId('a');
    expect(within(viewA).getAllByTestId('z')).toHaveLength(1);
    const viewB = wrapper.getByTestId('b');
    expect(() => within(viewB).getByTestId('z')).toThrow();
  });
});
