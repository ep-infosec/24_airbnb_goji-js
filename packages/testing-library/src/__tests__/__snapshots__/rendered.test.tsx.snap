// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendered debug 1`] = `
"[36m<view[39m
  [33mtestID[39m=[32m\\"a\\"[39m
[36m>[39m
  [36m<button[39m
    [33monTap[39m=[32m\\"[Function: onTap]\\"[39m
  [36m>[39m
    [0mhello[0m
  [36m</button>[39m
  [36m<text[39m
    [33mclassName[39m=[32m\\"a b c\\"[39m
  [36m>[39m
    [0mworld[0m
  [36m</text>[39m
[36m</view>[39m"
`;
